src/core/worker.js

```diff
@@ -565,6 +578,19 @@ class WorkerMessageHandler {
           pdfManager.ensureDoc("startXRef"),
         ];
 
+        for (const [pageIndex, annotations] of newAnnotationsByPage) {
+          promises.push(
+            pdfManager.getPage(pageIndex).then(page => {
+              const task = new WorkerTask(`Save (editor): page ${pageIndex}`);
+              return page
+                .saveNewAnnotations(handler, task, annotations)
+                .finally(function () {
+                  finishWorkerTask(task);
+                });
+            })
+          );
+        }
+
```

Comment:
As mentioned previously, this doesn't really make sense in XFA-documents (since it won't work) and you may want to down into the `!isPureXfa`-branch below.